modules/library/referencing/src/main/java/org/geotools/referencing/DefaultAuthorityFactory.java

```diff
@@ -80,36 +84,35 @@ private static AbstractAuthorityFactory getBackingFactory(final boolean longitud
         }
 
         Collection<CRSAuthorityFactory> factories =
-                ReferencingFactoryFinder.getCRSAuthorityFactories(hints);
+                new ArrayList<>(ReferencingFactoryFinder.getCRSAuthorityFactories(hints));
         if (Boolean.TRUE.equals(hints.put(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.FALSE))) {
             /*
              * If hints contain a requirement for "longitude first", then we may loose some
              * authorities like "URN:OGC:...". Search again without such requirement and add
              * any new authorities found.
              */
-            factories = new ArrayList<>(factories);
-            final Set<Citation> authorities = new LinkedHashSet<>();
-            for (final CRSAuthorityFactory factory : factories) {
-                authorities.add(factory.getAuthority());
-            }
-            searchNews:
-            for (final CRSAuthorityFactory factory :
-                    ReferencingFactoryFinder.getCRSAuthorityFactories(hints)) {
-                final Citation authority = factory.getAuthority();
-                if (authorities.contains(authority)) {
-                    continue;
-                }
-                for (final Citation check : authorities) {
-                    if (Citations.identifierMatches(authority, check)) {
-                        continue searchNews;
-                    }
-                }
-                factories.add(factory);
+
```

Comment:
So... this change is giving me nightmares. Not because I think it's wrong, per se, but because the referencing subsystem is known for reconfiguring itself dynamically, scanning for factories makes new factories pop up or re-order themselves (don't ask, I don't remember why this happens). That's why there was this loop, which would otherwise seem nonsensical.

That said, all tests are passing. On the other end, the whole ticket is addressing an IDE setup deficiency, not an actual runtime issue.
I'm going to wait until 27-RC is cut, and then merge when main becomes 28.x. This should give us six months to see f there are issues with the change that tests are not finding, and revert before this gets in the user's face.